From efd7501c7e9e0c14a3e7257838930893b77a868f Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Fri, 12 Aug 2022 17:04:04 +0800
Subject: [PATCH 38/46] Revert "drm/komeda: return early if
 drm_universal_plane_init() fails."

This reverts commit aea748501d0923c5d793a86aa39e65b3538dac64.
---
 drivers/gpu/drm/arm/display/komeda/komeda_plane.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
index a5f57b38d193..98e915e325dd 100644
--- a/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
+++ b/drivers/gpu/drm/arm/display/komeda/komeda_plane.c
@@ -274,10 +274,8 @@ static int komeda_plane_add(struct komeda_kms_dev *kms,
 
 	komeda_put_fourcc_list(formats);
 
-	if (err) {
-		kfree(kplane);
-		return err;
-	}
+	if (err)
+		goto cleanup;
 
 	drm_plane_helper_add(plane, &komeda_plane_helper_funcs);
 
-- 
2.27.0

